easybuild/easyblocks/generic/craytoolchain.py

```diff
@@ -0,0 +1,66 @@
+##
+# Copyright 2015-2015 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for installing Cray toolchains, implemented as an easyblock
+
+@author: Kenneth Hoste (Ghent University)
+"""
+
+from easybuild.easyblocks.generic.bundle import Bundle
+
+
+KNOWN_PRGENVS = ['PrgEnv-cray', 'PrgEnv-gnu', 'PrgEnv-intel', 'PrgEnv-pgi']
+
+
+class CrayToolchain(Bundle):
+    """
+    Compiler toolchain: generate module file only, nothing to build/install
+    """
+    def make_module_dep(self):
+        """
+        Generate load/swap statements for dependencies in the module file
+        """
+        # build dict with info for 'module swap' statements for dependencies
+        swap_info = {}
+        for dep in self.toolchain.dependencies:
+            mod_name = dep['full_mod_name']
+            if not mod_name.startswith('PrgEnv-'):
+                depname = '/'.join(mod_name.split('/')[:-1])
+                swap_info.update({mod_name: depname})
```

Comment:
this is creating a map like:

```
{
    'fftw/3.3.4.2': 'fftw',
    'cray-libsci/13.0.4': 'cray-libsci',
}
```

We need to know what should be unloaded/swapped out, i.e. the 'versionless module name'